File: NETMouse - .NET release/Examples/C#/Matrices_6.cs
using ABCNET.Extensions;
using ABCNET.Utils;
using System;

namespace TestProgram
{
    internal static class Program
    {
        public static int[] Count(this int[,] matrix)
        {
            if (matrix == null)
                throw new ArgumentNullException("matrix");

            int[] result = new int[matrix.GetLength(0)];
            int rowsCount = matrix.GetLength(0);
            int colsCount = matrix.GetLength(1);
            for (int i = 0; i < rowsCount; i++)
                for (int j = 0; j < colsCount; j++)
                    if (matrix[i, j] > 0)
                        result[i] += 1;

            return result;
        }

        private static void Main(string[] args)
        {
            Matr.ReadInteger(Base.ReadInteger("N:"), Base.ReadInteger("M:"), "Элемент ({0}, {1})-ый:").Count().Println();
        }
    }
}

File: NETMouse - .NET release/Examples/PascalABC.NET/Matrices_6.pas
{$reference ABCNET.dll}
uses ABCNET.Utils;
uses ABCNET.Extensions;
uses System;

function Count(self: array [,] of integer): array of integer; extensionmethod;
begin
  if self = nil then
    raise new ArgumentNullException('self');

  result := new integer[self.GetLength(0)];
  var rowsCount := self.GetLength(0);
  var colsCount := self.GetLength(1);
  for var i := 0 to rowsCount - 1 do
    for var j := 0 to colsCount - 1 do
      if self[i, j] > 0 then
        result[i] += 1;
end;

begin
  Matr.ReadInteger(BaseU.ReadInteger('N:'), BaseU.ReadInteger('M:'), 'Элемент ({0}, {1})-ый:').Count().Println();
  BaseU.ReadChar();
end.